packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -0,0 +1,796 @@
+            self.add_new_line(1)
+
+        for param_num in range(0, len(self.parameters)):
+            if self.parameters[param_num]:
+                self.parameters[param_num].to_token()
+            if param_num == 0:
+                self.add_first_line()
+            self.add_new_line()
+
+            # Add in parameter tokens
+            if self.parameters[param_num]:
+                self.add_whitespace(4)
+                for t in self.parameters[param_num].get_tokens():
+                    self.add_token(t)
+                    self.overview_tokens.append(t)
+
+            # Add in comma before the next parameter
+            if param_num + 1 in range(0, len(self.parameters)):
+                self.parameters[param_num + 1]
+                self.add_punctuation(",")
+                self.overview_tokens.append(Token(", ", TokenKind.Text))
+
+            # Create a new line for the next operation
+            else:
+                self.add_new_line()
+                self.add_whitespace(3)
+                self.overview_tokens.append(Token(")", TokenKind.Text))
+                self.add_punctuation(")")
+                self.add_new_line(1)
+
+                self.add_token(Token(kind=TokenKind.StartDocGroup))
+
+    def to_json(self):
+        obj_dict = {}
+        self.to_token()
+        for key in OP_FIELDS:
+            obj_dict[key] = self.__dict__[key]
+        return obj_dict
+
+
+class LLCParameterView(FormattingClass):
+    def __init__(
+        self,
+        param_name,
+        param_type,
+        namespace,
+        json_request=None,
+        default=None,
+        required=False,
+    ):
+        self.name = param_name
+        self.type = param_type
+        self.default = default
+        self.required = required
+        self.Tokens = []
+        self.overview_tokens = []
+        self.json_request = json_request
+        self.namespace = namespace
+
+    @classmethod
+    def from_yaml(cls, yaml_data: Dict[str, Any], i, name):
+        required = True
+        default = None
+        json_request = {}
+        if yaml_data.get("signatureParameters"):
+            default = yaml_data["signatureParameters"][i]["schema"].get("defaultValue")
+            param_name = yaml_data["signatureParameters"][i]["language"]["default"][
+                "name"
+            ]
+            if yaml_data["signatureParameters"][i]["schema"]["type"] == "object":
+                param_type = get_type(
+                    yaml_data["signatureParameters"][i]["schema"]["properties"][0][
+                        "schema"
+                    ]
+                )
+            else:
+                param_type = get_type(yaml_data["signatureParameters"][i]["schema"])
+            if param_name == "body":
+                try:
+                    param_name = yaml_data["signatureParameters"][i]["schema"][
+                        "properties"
+                    ][0]["serializedName"]
+                except:
```

Comment:
not crucial, but I think we could narrow down the exception type to catch here to KeyError or IndexError.

Reply:
made issue